velocity/tool/src/java/org/sakaiproject/cheftool/VelocityPortletPaneledAction.java

```diff
@@ -343,6 +345,18 @@ protected void toolModeDispatch(String methodBase, String methodExt, HttpServlet
             context.put("language",languageCode);
             context.put("dir", rl.getOrientation(locale));
 
+			String thisUser = SessionManager.getCurrentSessionUserId();
+			PreferencesService preferencesService = ComponentManager.get(PreferencesService.class);
+
+			Preferences prefs = preferencesService.getPreferences(thisUser);
+
+			String userTheme = "sakaiUserTheme-notSet";
+			if ( prefs != null ) {
+				userTheme = StringUtils.defaultIfEmpty(prefs.getProperties(org.sakaiproject.user.api.PreferencesService.USER_SELECTED_UI_THEME_PREFS).getProperty("theme"), "sakaiUserTheme-notSet");
+			}
+
+			context.put("userTheme", userTheme);
+
```

Comment:
I'd suggest we wrap all of this with a condition `if(sakaiThemesEnabled)` that checks for the "portal.themes" property: `sakaiThemesEnabled = ServerConfigurationService.getBoolean("portal.themes", true);` since there is no need to check the preference if themes are disabled in the instance.

Reply:
Check out the latest